src/pykiso/lib/auxiliaries/udsaux/uds_auxiliary.py

```diff
@@ -224,6 +230,34 @@ def write_data(
             log.error("No uds config found")
             return
 
+    def sender_run(self, period: int, stop_event: threading.Event) -> None:
+        """send tester present at defined period until stopped
+
+        :param period: period in seconds to use for the cyclic sending of tester present
+        :param stop_event: event to set to stop the sending of tester present
+        """
+        while not stop_event.is_set():
+            self.send_uds_raw(UDSCommands.TesterPresent.TESTER_PRESENT_NO_RESPONSE)
+            time.sleep(period)
+
+    @contextmanager
+    def tester_present_sender(self, period: int = 4) -> None:
+        """Context manager that continuously sends tester present messages via UDS
+
+        :param period: period in seconds to use for the cyclic sending of tester present
+        """
+        stop_event = threading.Event()
+        sender = threading.Thread(
+            name="TesterPresentSender",
+            target=self.sender_run,
+            args=(period, stop_event),
+        )
+        try:
+            yield sender.start()
```

Comment:
You could yield the thread instance after starting it so the user can eventually stop and restart it (you'll need to check for the is_alive method when exiting the context)

Reply:
Done